recipes/qt5/bld.bat

```diff
@@ -1,104 +1,137 @@
+:: Windows SDK, but conda-build should take care of that probably??
+:: It requires TARGET_ARCH to be one of x64 or x86.  On conda-forge
+:: this script is run *before* bld.bat.
+if "%ARCH%" == "64" (
+  set TARGET_ARCH=x64
+) else (
+  set TARGET_ARCH=x86
+)
+call %RECIPE_DIR%\obvci_appveyor_python_build_env.cmd
+
+if "%DXSDK_DIR%" == "" (
+  echo You do not appear to have the DirectX SDK installed.  Please get it from
+  echo    https://www.microsoft.com/en-us/download/details.aspx?id=6812
+  echo and try this build again.  If you have installed it, and are still seeing
+  echo this message, please open a new console to refresh your environment variables.
+  exit /b 1
 )
 
+:: Webengine requires either working OpenGL drivers or
+:: Angle (therefore DirectX >= 11). This works on some
+:: VMs and not others.  Windows 7 VirtualBox instantly
+:: aborts when loading Spyder sSo we've had to disable
+:: it globally.
+:: Using Mesa from MSYS2 is mentioned as a workaround:
+::  http://wiki.qt.io/Cross-compiling-Mesa-for-Windows
+:: `set QT_OPENGL=software` forces Qt5 to use that but
+:: but when I tried it it was too buggy; Spyder crashed a
+:: little bit later, though it worked for Carlos.
+set AVOID_WEBENGINE=yes
+set WEBTHING=qtwebengine
+if %VS_MAJOR% LSS 14 (
+  set WEBTHING=qtwebkit
+)
+if "%AVOID_WEBENGINE%" == "yes" (
+  set WEBTHING=qtwebkit
+)
+
+:: Shorten our build path as much as possible
+set _BLDTMP=C:\qt5-%ARCH%-%CONDA_PY%
 if "%DIRTY%" == "" (
-    :: Shorten our build path as much as possible
-    :: cd ..
+  if "%WEBTHING%" == "qtwebkit" (
     curl -LO "http://download.qt.io/community_releases/%SHORT_VERSION%/%PKG_VERSION%/qtwebkit-opensource-src-%PKG_VERSION%.tar.xz"
     if errorlevel 1 exit 1
-    7za x -so qtwebkit-opensource-src-%PKG_VERSION%.tar.xz | 7za x -si -aoa -ttar
+    7za x -so qtwebkit-opensource-src-%PKG_VERSION%.tar.xz | 7za x -si -aoa -ttar > NUL 2>&1
     if errorlevel 1 exit 1
-    MOVE qtwebkit-opensource-src-%PKG_VERSION% qtwebkit
-    mkdir C:\qt5
-    :: Copy all files except bld.bat, which needs to stay in place while script runs
-    robocopy %SRC_DIR%\ C:\qt5\ *.* /E /move /NFL /NDL /xf bld.bat
+    move qtwebkit-opensource-src-%PKG_VERSION% qtwebkit
+  )
+  if exist %_BLDTMP% (
+    rmdir /s /q %_BLDTMP%
+  )
+  mkdir %_BLDTMP%
+  :: Copy all files except bld.bat, which needs to stay in place while script runs
+  robocopy %SRC_DIR%\ %_BLDTMP%\ *.* /E /NFL /NDL /xf bld.bat
 )
-cd C:\qt5
+pushd %_BLDTMP%
 
-:: set path to find resources shipped with qt-5
+:: set path to find resources shipped with qt-5 (and also Ruby for qtwebkit)
 :: see http://doc-snapshot.qt-project.org/qt5-5.4/windows-building.html
-set "PATH=%CD%\qtbase\bin;%CD%\gnuwin32\bin;%PATH%"
-
-:: Install a custom python 27 environment for us, to use in building, but avoid feature activation
-conda create -y -n python27_qt5_build python=2.7
+set "PATH=%CD%\qtbase\bin;%CD%\gnuwin32\bin;C:\Ruby23\bin;%PATH%"
```

Comment:
* Could you mention in `Notes.md` that you need to install Ruby before running this?
* Why `gnuwin32` is needed here? Don't you have all that you need from the MSYS2 channel?